supabase/migrations/20260815184404_create_knowledge_library.sql

```
@@ -0,0 +1,145 @@
+    )
+    or (
+      status = 'failed'
+      and size_bytes is not null
+      and indexed_at is null
+      and last_error_code is not null
+    )
+  )
+);
+
+create index knowledge_sources_owner_status_created_idx
+  on public.knowledge_sources (created_by, status, created_at desc, id desc);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)